components/operator/internal/core/controllers.go

```diff
@@ -15,7 +16,15 @@ type ObjectController[T client.Object] func(ctx Context, req T) error
 
 func ForObjectController[T v1beta1.Object](controller ObjectController[T]) ObjectController[T] {
 	return func(ctx Context, object T) error {
+
+		logger, err := logr.FromContext(ctx)
```

Comment:
Why introducing a new library for logging ?

Reply:
Provided by the kubebuilder framework ;)